sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseUriInsteadOfString.cs

```diff
@@ -106,6 +112,34 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                     }
                 },
                 SyntaxKind.PropertyDeclaration);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var invokedMethod = c.SemanticModel.GetSymbolInfo(c.Node).Symbol as IMethodSymbol;
+                    if (invokedMethod == null)
+                    {
+                        return;
+                    }
+
+                    if (invokedMethod.IsInType(KnownType.System_Uri))
+                    {
+                        return;
+                    }
+
+                    var stringUrlParams = GetStringUrlParamIndexes(invokedMethod);
+                    if (!stringUrlParams.Any())
```

Comment:
Keep in mind you should use `stringUrlParams.Count == 0` as a replacement for `!stringUrlParams.Any()`.

Also I think we should measure the difference. Property access is indeed faster, but method call + cast is not slow as well....

As a functional alternative, we could add a few extension methods for IEnumerable<T> and ICollection<T>:
- IsEmpty/Empty/None/etc. with no arguments and Func<T, bool>
- HasCount(int) and HasCount(int, Func<T, bool>) 

I browsed a couple of LINQ extension libraries this morning, I think we could "borrow" some methods :) WDYT?